### DISCUSSION TITLE
Upgrade notification banner

### DISCUSSION THREAD

**Discussion by barmalej1** [netbox-community/netbox#8800] [Q&A]
Does anybody know how to hide it? I set RELEASE_CHECK_URL to None in netbox.env file, but it didn't have any effect

**Comment by jeremystretch** [comment]
Did you restart the NetBox service after modifying the configuration?

**Reply by barmalej1** [reply]
Multiple times. I am running containerized version of NetBox if it makes any difference.

**Reply by kkthxbye-code** [reply]
You have to remove the "latest_release" key from the redis cache. Clearing this values should probably be added here @jeremystretch:

https://github.com/netbox-community/netbox/blob/f94c1e91ea20c805528125407193371476ccbcd8/netbox/extras/management/commands/housekeeping.py#L100-L102

**Reply by barmalej1** [reply]
How can I do it if I am running NetBox in containers? I tried to clear redis cache and restart the app, but the banner is still there

**Reply by kkthxbye-code** [reply]
Try the following:

Start an nbshell (see: https://github.com/netbox-community/netbox-docker/wiki/Troubleshooting#netbox-shell-nbshell) 

Execute the following:

```
from django.core.cache import cache
cache.delete("latest_release")
```

**Reply by barmalej1** [reply]
That didn't work either. BTW I couldn't start nbshell when RELEASE_CHECK_URL was set to None. I was getting this: 
`django.core.exceptions.ValidationError: ['RELEASE_CHECK_URL must be a valid API URL. Example: https://api.github.com/repos/netbox-community/netbox']`
I put a random but properly formatted URL there first and then I was able to ran the command above

**Reply by candlerb** [reply]
Did you set it to `None`, or `"None"` ??

```
RELEASE_CHECK_URL = None      # correct
RELEASE_CHECK_URL = "None"    # wrong!
```

The error you cite is only raised if the value is not empty; `"None"` is non-empty string that it tries to parse as a URL.

```
if RELEASE_CHECK_URL:
    validator = URLValidator(
        message=(
            "RELEASE_CHECK_URL must be a valid API URL. Example: "
            "https://api.github.com/repos/netbox-community/netbox"
        )
    )
    ...
```

EDIT: if you're doing this via container environment variables rather than in the config file then you don't have a choice (everything is a string), so simply leave `RELEASE_CHECK_URL` unset in the environment. Netbox-docker has:

```
./configuration/configuration.py:RELEASE_CHECK_URL = environ.get('RELEASE_CHECK_URL', None)
```

i.e. if it's not present then `None` is used as the value.

(Note: I'm looking at current code here, so if you're running an older version of netbox-docker/netbox then it may be different)

**Reply by barmalej1** [reply]
I set it to None without quotes as was described in the document. Right now I have some random URL there. In configuration.py `RELEASE_CHECK_URL` is set to `environ.get('RELEASE_CHECK_URL', None)`.
If it makes any difference, I have version 3.1.6, not the very latest but close.

**Reply by candlerb** [reply]
Yes, but where did you set it?  If you had set it directly in configuration.py then you'd set

```
RELEASE_CHECK_URL = None
```

However if you're using netbox-doccker, and you set an environment variable `RELEASE_CHECK_URL=None`, then this is a string value, so it's equivalent to

```
RELEASE_CHECK_URL = "None"
```

The solution is *not to set the environment variable at all*. The expression `environ.get('RELEASE_CHECK_URL', None)` in python means:

> give me the value of the environment variable `RELEASE_CHECK_URL`, but if it's not present, return `None`

(referring to the actual value `None` not the string `"None"`)

Confusing I know, but "the document" you are referring to is probably the documentation for Netbox, not netbox-docker.

**Reply by barmalej1** [reply]
Success! I added `RELEASE_CHECK_URL = None` to configuration/extra.py and deleted cache from nbshell one more time

Thanks a lot!
